### ISSUE TITLE
Fix Scene.visibleObjectIds update after unloading a model

### ISSUE THREAD

**Issue by xeolabs** [xeokit/xeokit-sdk#1086]
While unloading models, the visibleObjects collection on the scene does not refresh correctly (object ids will be added to this collection instead of removed). 
Objects (all loaded objects) collection is working correctly.

**Comment by LukasKairnial** [comment]
Hi there,

I have stumbled across this bug this afternoon.
It's not only linked to visibleObjects, but also, at least : 

- Xrayed objects
- Selected objects
- Highlighted objects
- Offseted objects

The bug is linked to a misuse of respective `Scene._objectXXXUpdated` functions.
[I have drafted a quick fix](https://github.com/xeokit/xeokit-sdk/commit/1d280bf0c14579cb6d47db861512831c834e7c56) which fixed all memory leaks in my scenes, but change behaviour on notifications and breaks `Scene.numOffsetObjects` (I'm a bit shy to modify these function behaviours since I'm not sure what could be impacted).
